requirements.txt

```diff
@@ -1,8 +1,10 @@
-ansible-core
+ansible-core < 2.17.0
```

Comment:
yeah there is a syntax error in 2.17 when ansible try to talk with rockylinux8 environment
I leave the detail here: https://github.com/vexxhost/ansible-collection-kubernetes/pull/119#issuecomment-2124841633

So IMO, we either constrain this version or drop rockylinux8 from CI

Reply:
In RHEL world this can/should be fixed by installing python3x using dnf and switched from default system Python using 'update-alternatives'. Distro's Python version will stay 3.6 until EOL.

Reply:
So the concern I have is we either support default rockylinux8 or ask to switch the python in rockylinux8 to fit this collection needs

I think we can use update-alternatives in real cases but not that handy in CI. Just make a patch to reinstall python to 3.9 so we don't need to touch ansible-core requirement.

Reply:
unfortunately can not change python version from molecule prepare stage as it failed on Gethering fact for prepare
```
PLAY [Prepare] *****************************************************************
  
  TASK [Gathering Facts] *********************************************************
  fatal: [instance]: FAILED! => {"ansible_facts": {}, "changed": false, "failed_modules": {"ansible.legacy.setup": {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python3"}, "exception": "Traceback (most recent call last):\n  File \"<stdin>\", line 107, in <module>\n  File \"<stdin>\", line [99](https://github.com/vexxhost/ansible-collection-kubernetes/actions/runs/9202557352/job/25312618142?pr=119#step:6:105), in _ansiballz_main\n  File \"<stdin>\", line 44, in invoke_module\n  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n  File \"<frozen importlib._bootstrap>\", line 951, in _find_and_load_unlocked\n  File \"<frozen importlib._bootstrap>\", line 894, in _find_spec\n  File \"<frozen importlib._bootstrap_external>\", line 1157, in find_spec\n  File \"<frozen importlib._bootstrap_external>\", line 1131, in _get_spec\n  File \"<frozen importlib._bootstrap_external>\", line 1112, in _legacy_get_spec\n  File \"<frozen importlib._bootstrap>\", line 441, in spec_from_loader\n  File \"<frozen importlib._bootstrap_external>\", line 544, in spec_from_file_location\n  File \"/tmp/ansible_ansible.legacy.setup_payload_8qv9jvhz/ansible_ansible.legacy.setup_payload.zip/ansible/module_utils/basic.py\", line 5\nSyntaxError: future feature annotations is not defined\n", "failed": true, "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 107, in <module>\n  File \"<stdin>\", line 99, in _ansiballz_main\n  File \"<stdin>\", line 44, in invoke_module\n  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n  File \"<frozen importlib._bootstrap>\", line 951, in _find_and_load_unlocked\n  File \"<frozen importlib._bootstrap>\", line 894, in _find_spec\n  File \"<frozen importlib._bootstrap_external>\", line 1157, in find_spec\n  File \"<frozen importlib._bootstrap_external>\", line 1131, in _get_spec\n  File \"<frozen importlib._bootstrap_external>\", line 1112, in _legacy_get_spec\n  File \"<frozen importlib._bootstrap>\", line 441, in spec_from_loader\n  File \"<frozen importlib._bootstrap_external>\", line 544, in spec_from_file_location\n  File \"/tmp/ansible_ansible.legacy.setup_payload_8qv9jvhz/ansible_ansible.legacy.setup_payload.zip/ansible/module_utils/basic.py\", line 5\nSyntaxError: future feature annotations is not defined\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}}, "msg": "The following modules failed to execute: ansible.legacy.setup\n"}
```
let me revert https://github.com/vexxhost/ansible-collection-kubernetes/pull/119/commits/7677c98ff5f675da2e7a57088e81143ff5435a59 for now